packages/repack/android/src/main/java/com/callstack/repack/RetryInterceptor.kt

```diff
@@ -0,0 +1,39 @@
+package com.callstack.repack
+
+import okhttp3.Interceptor
+import okhttp3.Response
+import java.io.IOException
+
+class RetryInterceptor(config: ScriptConfig) : Interceptor {
+    private val maxRetries = config.retry
+    private val retryDelayMillis = config.retryDelay
+
+    override fun intercept(chain: Interceptor.Chain): Response {
+        var retryCount = 0
+        var response: Response? = null
+        var lastException: IOException? = null
+
+        do {
+            try {
+                response?.close() // Ensure previous response is closed before retrying
+                response = chain.proceed(chain.request())
+                // If the response is successful, break the loop
+                if (response.isSuccessful) {
+                    return response
+                }
+            } catch (e: IOException) {
+                lastException = e
+            }
+
+            retryCount++
+
+            if (retryCount <= maxRetries) {
+                // Delay before retrying
+                Thread.sleep(retryDelayMillis.toLong())
```

Comment:
putting a thread to sleep might block some other work, it would be best to schedule the the retry rather than sleep